spark-connector/src/main/java/com/datastrato/gravitino/spark/catalog/GravitinoCatalog.java

```diff
@@ -0,0 +1,232 @@
+  // The Gravitino catalog client to do schema operations.
+  protected Catalog gravitinoCatalogClient;
+
+  private final String metalakeName;
+  private String catalogName;
+  private final GravitinoCatalogManager gravitinoCatalogManager;
+
+  public GravitinoCatalog() {
+    gravitinoCatalogManager = GravitinoCatalogManager.get();
+    metalakeName = gravitinoCatalogManager.getMetalakeName();
+  }
+
+  @Override
+  public Identifier[] listTables(String[] namespace) throws NoSuchNamespaceException {
+    throw new NotSupportedException("Doesn't support listing table");
+  }
+
+  @Override
+  public Table createTable(
+      Identifier ident, Column[] columns, Transform[] partitions, Map<String, String> properties)
+      throws TableAlreadyExistsException, NoSuchNamespaceException {
+    throw new NotSupportedException("Doesn't support creating table");
+  }
+
+  // Will create a catalog specific table by invoking createSparkTable()
+  @Override
+  public Table loadTable(Identifier ident) throws NoSuchTableException {
+    throw new NotSupportedException("Doesn't support loading table");
+  }
+
+  @SuppressWarnings("deprecation")
+  @Override
+  public Table createTable(
+      Identifier ident, StructType schema, Transform[] partitions, Map<String, String> properties)
+      throws TableAlreadyExistsException, NoSuchNamespaceException {
+    throw new NotSupportedException("Deprecated create table method");
+  }
+
+  @Override
+  public Table alterTable(Identifier ident, TableChange... changes) throws NoSuchTableException {
+    throw new NotSupportedException("Doesn't support altering table");
+  }
+
+  @Override
+  public boolean dropTable(Identifier ident) {
+    throw new NotSupportedException("Doesn't support drop table");
+  }
+
+  @Override
+  public void renameTable(Identifier oldIdent, Identifier newIdent)
+      throws NoSuchTableException, TableAlreadyExistsException {
+    throw new NotSupportedException("Doesn't support renaming table");
+  }
+
+  @Override
+  public void initialize(String name, CaseInsensitiveStringMap options) {
+    this.catalogName = name;
+    this.gravitinoCatalogClient = gravitinoCatalogManager.getGravitinoCatalogInfo(name);
+    String provider = gravitinoCatalogClient.provider();
+    Preconditions.checkArgument(provider != null, name + " catalog provider is null");
+    this.sparkCatalog = createAndInitSparkCatalog(provider, name, options);
+  }
+
+  @Override
+  public String name() {
+    return catalogName;
+  }
+
+  @Override
+  public String[][] listNamespaces() throws NoSuchNamespaceException {
+    NameIdentifier[] schemas =
+        gravitinoCatalogClient.asSchemas().listSchemas(Namespace.of(metalakeName, catalogName));
+    return Arrays.stream(schemas)
+        .map(schema -> new String[] {schema.name()})
+        .toArray(String[][]::new);
+  }
+
+  @Override
+  public String[][] listNamespaces(String[] namespace) throws NoSuchNamespaceException {
+    if (namespace.length == 0) {
```

Comment:
Optional: Should we use `checkArgument` here?

Reply:
done